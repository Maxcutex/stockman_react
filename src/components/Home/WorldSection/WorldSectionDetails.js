import React from 'react';
import WorldSectionShareBlock from './WorldSectionShareBlock';
import image from '../../../assets/images/home-3-330x220.jpg';

const WorldSectionDetails = props => {
    return props.activeNews.map(activeNew => {
        const {
            section,
            title,
            description,
            author: { first_name, last_name },
            date,
            id,
            visual_news,
        } = activeNew;
        return (
            <div
                className="col-xs-12 col-sm-6 col-md-4 col-lg-3 isotope-item"
                data-filter={section}
                key={id}
            >
                <div className="post-type-2">
                    <div className="img-block">
                        <a href="post.html">
                            <img
                                // src={visual_news[0].image_file || image}
                                src={image}
                                width="330"
                                height="220"
                                alt=""
                            />
                        </a>
                        <ul className="tag-list">
                            <li>
                                <a href="post.html">{section}</a>
                            </li>
                        </ul>
                    </div>
                    <div className="caption">
                        <h5>
                            <a href="post.html">{title}</a>
                        </h5>
                        <p>{description}</p>
                        <div className="bottom-block">
                            <ul className="meta-list">
                                <li>
                                    by{' '}
                                    <a href="post.html">{`${first_name} ${last_name}`}</a>
                                </li>
                                <li>
                                    <a href="post.html">{date}</a>
                                </li>
                            </ul>
                            <WorldSectionShareBlock />
                        </div>
                    </div>
                </div>
            </div>
        );
    });
};
export default WorldSectionDetails;
