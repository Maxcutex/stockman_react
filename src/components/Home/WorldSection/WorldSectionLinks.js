import React from "react"
import {Link} from "react-router-dom";


export default function WorldSectionLinks({changeSection}) {
    return (
        <ul
            className="isotope-filters-list"
            id="isotope-filters"
        >
            <li>
                <Link
                    to=""
                    onClick={() =>
                        changeSection('World')
                    }
                >
                    World{' '}
                </Link>
            </li>
            <li>
                <Link
                    to=""
                    onClick={() =>
                        changeSection(
                            'Business'
                        )
                    }
                >
                    Business{' '}
                </Link>
            </li>
            <li>
                <Link
                    to=""
                    onClick={() =>
                        changeSection(
                            'Politics'
                        )
                    }
                >
                    Politics{' '}
                </Link>
            </li>
            <li>
                <a
                    data-isotope-filter="Category 4"
                    data-isotope-group="gallery"
                    href="/"
                >
                    Economy{' '}
                </a>
            </li>
            <li>
                <a
                    data-isotope-filter="Category 5"
                    data-isotope-group="gallery"
                    href="/"
                >
                    Markets{' '}
                </a>
            </li>
            <li>
                <a
                    data-isotope-filter="Category 6"
                    data-isotope-group="gallery"
                    href="/"
                >
                    Art{' '}
                </a>
            </li>
            <li>
                <a
                    data-isotope-filter="Category 7"
                    data-isotope-group="gallery"
                    href="/"
                >
                    Life{' '}
                </a>
            </li>
        </ul>
    )
}