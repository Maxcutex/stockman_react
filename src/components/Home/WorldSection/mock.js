export const news =  [
    {
        id: 1,
        section: 'Economics',
        title: 'news title1',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
    },
    {
        id: 2,
        section: 'Business',
        title: 'news Business',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
    },
    {
        section: 'Business',
        title: 'news Business2',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 3,
    },
    {
        section: 'Business',
        title: 'news Business3',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 4,
    },
    {
        section: 'Politics',
        title: 'news Politics',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 5,
    },
    {
        section: 'Business',
        title: 'news title2',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 6,
    },
    {
        section: 'Politics',
        title: 'news title3',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 7,
    },
    {
        section: 'World',
        title: 'news title1',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 8,
    },
    {
        section: 'World',
        title: 'news title2',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 9,
    },
    {
        section: 'World',
        title: 'news title3',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 10,
    },
    {
        id: 11,
        section: 'World',
        title: 'news Business',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
    },
];

export const activeNews = [
    {
        section: 'World',
        title: 'news title1',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 8,
    },
    {
        section: 'World',
        title: 'news title2',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 9,
    },
    {
        section: 'World',
        title: 'news title3',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
        id: 10,
    },
    {
        id: 11,
        section: 'World',
        title: 'news Business',
        description: 'this is the description',
        author: 'Jon Snow',
        date: '28-01-2019',
    },
];