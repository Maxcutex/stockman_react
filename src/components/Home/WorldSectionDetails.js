import React from 'react';
import { Link } from 'react-router-dom';

const WorldSectionDetails = props => {

    return props.activeNews.map(activeNew => {
        const { section, title, description, author, date,id } = activeNew;
        return (
            
                <div
                    className="col-xs-12 col-sm-6 col-md-4 col-lg-3 isotope-item"
                    data-filter={section} key={id}
                >
                    <div className="post-type-2">
                        <div className="img-block">
                            <a href="post.html">
                                <img
                                    src="images/home-3-330x220.jpg"
                                    width="330"
                                    height="220"
                                    alt=""
                                />
                            </a>
                            <ul className="tag-list">
                                <li>
                                    <a href="post.html">{section}</a>
                                </li>
                            </ul>
                        </div>
                        <div className="caption">
                            <h5>
                                <a href="post.html">{title}</a>
                            </h5>
                            <p>{description}</p>
                            <div className="bottom-block">
                                <ul className="meta-list">
                                    <li>
                                        by <a href="post.html">{author}</a>
                                    </li>
                                    <li>
                                        <a href="post.html">{date}</a>
                                    </li>
                                </ul>
                                <div className="share-block">
                                    <div className="soc-icon">
                                        <Link
                                            className="icon fa-twitter"
                                            to="/"
                                        >
                                            {' '}
                                        </Link>
                                        <Link
                                            className="icon fa-facebook"
                                            to="/"
                                        />
                                        <Link
                                            className="icon fa-google-plus"
                                            to="/"
                                        />
                                        <Link
                                            className="icon fa-youtube-play"
                                            to="/"
                                        />
                                    </div>
                                    <span className="icon fa-share-alt" />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
           
        );
    });
};
export default WorldSectionDetails;
