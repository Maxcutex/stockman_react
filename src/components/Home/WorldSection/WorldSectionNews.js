import React from 'react';
import WorldSectionDetails from './WorldSectionDetails';
import { AnimateOnChange } from 'react-animation'

export default function WorldSectionNews({ activeNews }) {
    return (
        <div className="cell-xs-12">
            <div
                className="row"
                style={{
                    marginLeft: 'auto',
                    marginRight: 'auto',
                    position: 'relative',
                }}
            >
                <div
                    className="isotope isotope--loaded"
                    data-isotope-layout="fitRows"
                    data-isotope-group="gallery"
                ><AnimateOnChange animationIn="bounceIn" animationOut="bounceOut" style={{width:'100%'}}>
                    <div className="row">
                       
                        <WorldSectionDetails activeNews={activeNews} />
                    </div>
                    </AnimateOnChange>
                </div>
            </div>
        </div>
    );
}
