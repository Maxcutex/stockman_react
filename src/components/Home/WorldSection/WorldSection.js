import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchNews, fetchWorldNews } from '../../../actions/newsAction';

import WorldSectionLinks from './WorldSectionLinks';
import WorldSectionNews from './WorldSectionNews';
import Loader from '../../Loader/Loader';

class WorldSection extends Component {
    state = {
        activeNews: [],
        showInMobileForm: false
    };
    componentDidMount() {
        this.props.fetchWorldNews();
    }

    componentDidUpdate(prevProps) {
        if (this.props.worldNews==null || !this.props.worldNews.length) {
            this.props.fetchWorldNews();
        }

        if (this.props !== prevProps && this.props.worldNews) {
            this.setState({ activeNews: this.filterSection('World') });
        }
    }
    showMobileForm = e => {
        this.setState({ showInMobileForm: !this.state.showInMobileForm });
      };
    filterSection = section => {
        const results = this.props.worldNews;
        const newResultList = [];
        results.forEach(result => {
            const sectionName = result.category_news.filter(
                nws => nws.section_category.section_name === section
            );
            if (sectionName.length) newResultList.push(result);
        });

        return newResultList;
    };

    changeSection = (e, section) => {
        e.preventDefault();
        const newResultList = this.filterSection(section);
        this.setState({ activeNews: newResultList,showInMobileForm: !this.state.showInMobileForm  });
    };

    render() {
        if (this.props.isLoading) {
            return <Loader />;
        }
        let menuButtonClass = ['isotope-filters-toggle', 'isotope-filters-toggle-1', 'button'];
        if (this.state.showInMobileForm) {
            menuButtonClass.push('active');
          }

        return (
            <section className="section-sm bg-white">
                <div className="shell">
                    <div className="range">
                        <div className="cell-xs-12">
                            <div className="section-title">
                                <h3>Investing</h3>
                                <div className="isotope-filters isotope-filters-horizontal">
                                    <button
                                        className={menuButtonClass.join(' ')}
                                        data-custom-toggle="#isotope-filters"
                                        data-custom-toggle-disable-on-blur="true" onClick={this.showMobileForm.bind(
                                            this
                                        )}
                                    >
                                        Filter
                                        <span className="caret" />
                                    </button>
                                    <WorldSectionLinks showInMobileForm={this.state.showInMobileForm}
                                        changeSection={(e, section) =>
                                            this.changeSection(e, section)
                                        }
                                    />
                                </div>
                            </div>
                        </div>
                        <WorldSectionNews activeNews={this.state.activeNews} />
                    </div>
                </div>
            </section>
        );
    }
}

const mapStateToProps = ({ news }) => ({
    generalNews: news.generalNews,
    worldNews: news.worldNews,
    isLoading: news.generalNewsLoading,
});

export default connect(mapStateToProps, { fetchNews, fetchWorldNews })(
    WorldSection
);
