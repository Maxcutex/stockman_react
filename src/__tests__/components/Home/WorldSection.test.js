import React from 'react';
import { shallow } from 'enzyme';

import WorldSection from '../../../components/Home/WorldSection/WorldSection';
import WorldSectionDetails from '../../../components/Home/WorldSection/WorldSectionDetails';
describe('WorldSection Component', () => {
    let component;
    let props;
    beforeEach(() => {
        props = {
            activeNews: [
                {
                    section: 'World',
                    title: 'news title1',
                    description: 'this is the description',
                    author: 'Jon Snow',
                    date: '28-01-2019',
                    id: 8,
                },
            ],
        };
        component = shallow(<WorldSection />);
    });
    it('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });

    it('has sub component world section details', () => {
        const subComp = component.find(WorldSectionDetails);
        expect(subComp.length).toBe(1);
    });
    it('renders World Section Details without crashing', () => {
        let componentDetails = shallow(<WorldSectionDetails {...props} />);
    });
});
