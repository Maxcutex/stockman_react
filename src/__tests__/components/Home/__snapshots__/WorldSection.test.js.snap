// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorldSection Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorldSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="shell"
      >
        <div
          className="range"
        >
          <div
            className="cell-xs-12"
          >
            <div
              className="section-title"
            >
              <h3>
                World
              </h3>
              <div
                className="isotope-filters isotope-filters-horizontal"
              >
                <button
                  className="isotope-filters-toggle isotope-filters-toggle-1 button"
                  data-custom-toggle="#isotope-filters"
                  data-custom-toggle-disable-on-blur="true"
                >
                  Filter
                  <span
                    className="caret"
                  />
                </button>
                <WorldSectionLinks
                  changeSection={[Function]}
                />
              </div>
            </div>
          </div>
          <WorldSectionNews
            activeNews={
              Array [
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 8,
                  "section": "World",
                  "title": "news title1",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 9,
                  "section": "World",
                  "title": "news title2",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 10,
                  "section": "World",
                  "title": "news title3",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 11,
                  "section": "World",
                  "title": "news Business",
                },
              ]
            }
          />
        </div>
      </div>,
      "className": "section-sm bg-white",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="range"
        >
          <div
            className="cell-xs-12"
          >
            <div
              className="section-title"
            >
              <h3>
                World
              </h3>
              <div
                className="isotope-filters isotope-filters-horizontal"
              >
                <button
                  className="isotope-filters-toggle isotope-filters-toggle-1 button"
                  data-custom-toggle="#isotope-filters"
                  data-custom-toggle-disable-on-blur="true"
                >
                  Filter
                  <span
                    className="caret"
                  />
                </button>
                <WorldSectionLinks
                  changeSection={[Function]}
                />
              </div>
            </div>
          </div>
          <WorldSectionNews
            activeNews={
              Array [
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 8,
                  "section": "World",
                  "title": "news title1",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 9,
                  "section": "World",
                  "title": "news title2",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 10,
                  "section": "World",
                  "title": "news title3",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 11,
                  "section": "World",
                  "title": "news Business",
                },
              ]
            }
          />
        </div>,
        "className": "shell",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cell-xs-12"
            >
              <div
                className="section-title"
              >
                <h3>
                  World
                </h3>
                <div
                  className="isotope-filters isotope-filters-horizontal"
                >
                  <button
                    className="isotope-filters-toggle isotope-filters-toggle-1 button"
                    data-custom-toggle="#isotope-filters"
                    data-custom-toggle-disable-on-blur="true"
                  >
                    Filter
                    <span
                      className="caret"
                    />
                  </button>
                  <WorldSectionLinks
                    changeSection={[Function]}
                  />
                </div>
              </div>
            </div>,
            <WorldSectionNews
              activeNews={
                Array [
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 8,
                    "section": "World",
                    "title": "news title1",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 9,
                    "section": "World",
                    "title": "news title2",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 10,
                    "section": "World",
                    "title": "news title3",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 11,
                    "section": "World",
                    "title": "news Business",
                  },
                ]
              }
            />,
          ],
          "className": "range",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="section-title"
              >
                <h3>
                  World
                </h3>
                <div
                  className="isotope-filters isotope-filters-horizontal"
                >
                  <button
                    className="isotope-filters-toggle isotope-filters-toggle-1 button"
                    data-custom-toggle="#isotope-filters"
                    data-custom-toggle-disable-on-blur="true"
                  >
                    Filter
                    <span
                      className="caret"
                    />
                  </button>
                  <WorldSectionLinks
                    changeSection={[Function]}
                  />
                </div>
              </div>,
              "className": "cell-xs-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    World
                  </h3>,
                  <div
                    className="isotope-filters isotope-filters-horizontal"
                  >
                    <button
                      className="isotope-filters-toggle isotope-filters-toggle-1 button"
                      data-custom-toggle="#isotope-filters"
                      data-custom-toggle-disable-on-blur="true"
                    >
                      Filter
                      <span
                        className="caret"
                      />
                    </button>
                    <WorldSectionLinks
                      changeSection={[Function]}
                    />
                  </div>,
                ],
                "className": "section-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "World",
                  },
                  "ref": null,
                  "rendered": "World",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="isotope-filters-toggle isotope-filters-toggle-1 button"
                        data-custom-toggle="#isotope-filters"
                        data-custom-toggle-disable-on-blur="true"
                      >
                        Filter
                        <span
                          className="caret"
                        />
                      </button>,
                      <WorldSectionLinks
                        changeSection={[Function]}
                      />,
                    ],
                    "className": "isotope-filters isotope-filters-horizontal",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Filter",
                          <span
                            className="caret"
                          />,
                        ],
                        "className": "isotope-filters-toggle isotope-filters-toggle-1 button",
                        "data-custom-toggle": "#isotope-filters",
                        "data-custom-toggle-disable-on-blur": "true",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Filter",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "caret",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "span",
                        },
                      ],
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "changeSection": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeNews": Array [
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 8,
                  "section": "World",
                  "title": "news title1",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 9,
                  "section": "World",
                  "title": "news title2",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 10,
                  "section": "World",
                  "title": "news title3",
                },
                Object {
                  "author": "Jon Snow",
                  "date": "28-01-2019",
                  "description": "this is the description",
                  "id": 11,
                  "section": "World",
                  "title": "news Business",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="shell"
        >
          <div
            className="range"
          >
            <div
              className="cell-xs-12"
            >
              <div
                className="section-title"
              >
                <h3>
                  World
                </h3>
                <div
                  className="isotope-filters isotope-filters-horizontal"
                >
                  <button
                    className="isotope-filters-toggle isotope-filters-toggle-1 button"
                    data-custom-toggle="#isotope-filters"
                    data-custom-toggle-disable-on-blur="true"
                  >
                    Filter
                    <span
                      className="caret"
                    />
                  </button>
                  <WorldSectionLinks
                    changeSection={[Function]}
                  />
                </div>
              </div>
            </div>
            <WorldSectionNews
              activeNews={
                Array [
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 8,
                    "section": "World",
                    "title": "news title1",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 9,
                    "section": "World",
                    "title": "news title2",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 10,
                    "section": "World",
                    "title": "news title3",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 11,
                    "section": "World",
                    "title": "news Business",
                  },
                ]
              }
            />
          </div>
        </div>,
        "className": "section-sm bg-white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="range"
          >
            <div
              className="cell-xs-12"
            >
              <div
                className="section-title"
              >
                <h3>
                  World
                </h3>
                <div
                  className="isotope-filters isotope-filters-horizontal"
                >
                  <button
                    className="isotope-filters-toggle isotope-filters-toggle-1 button"
                    data-custom-toggle="#isotope-filters"
                    data-custom-toggle-disable-on-blur="true"
                  >
                    Filter
                    <span
                      className="caret"
                    />
                  </button>
                  <WorldSectionLinks
                    changeSection={[Function]}
                  />
                </div>
              </div>
            </div>
            <WorldSectionNews
              activeNews={
                Array [
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 8,
                    "section": "World",
                    "title": "news title1",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 9,
                    "section": "World",
                    "title": "news title2",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 10,
                    "section": "World",
                    "title": "news title3",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 11,
                    "section": "World",
                    "title": "news Business",
                  },
                ]
              }
            />
          </div>,
          "className": "shell",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cell-xs-12"
              >
                <div
                  className="section-title"
                >
                  <h3>
                    World
                  </h3>
                  <div
                    className="isotope-filters isotope-filters-horizontal"
                  >
                    <button
                      className="isotope-filters-toggle isotope-filters-toggle-1 button"
                      data-custom-toggle="#isotope-filters"
                      data-custom-toggle-disable-on-blur="true"
                    >
                      Filter
                      <span
                        className="caret"
                      />
                    </button>
                    <WorldSectionLinks
                      changeSection={[Function]}
                    />
                  </div>
                </div>
              </div>,
              <WorldSectionNews
                activeNews={
                  Array [
                    Object {
                      "author": "Jon Snow",
                      "date": "28-01-2019",
                      "description": "this is the description",
                      "id": 8,
                      "section": "World",
                      "title": "news title1",
                    },
                    Object {
                      "author": "Jon Snow",
                      "date": "28-01-2019",
                      "description": "this is the description",
                      "id": 9,
                      "section": "World",
                      "title": "news title2",
                    },
                    Object {
                      "author": "Jon Snow",
                      "date": "28-01-2019",
                      "description": "this is the description",
                      "id": 10,
                      "section": "World",
                      "title": "news title3",
                    },
                    Object {
                      "author": "Jon Snow",
                      "date": "28-01-2019",
                      "description": "this is the description",
                      "id": 11,
                      "section": "World",
                      "title": "news Business",
                    },
                  ]
                }
              />,
            ],
            "className": "range",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="section-title"
                >
                  <h3>
                    World
                  </h3>
                  <div
                    className="isotope-filters isotope-filters-horizontal"
                  >
                    <button
                      className="isotope-filters-toggle isotope-filters-toggle-1 button"
                      data-custom-toggle="#isotope-filters"
                      data-custom-toggle-disable-on-blur="true"
                    >
                      Filter
                      <span
                        className="caret"
                      />
                    </button>
                    <WorldSectionLinks
                      changeSection={[Function]}
                    />
                  </div>
                </div>,
                "className": "cell-xs-12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3>
                      World
                    </h3>,
                    <div
                      className="isotope-filters isotope-filters-horizontal"
                    >
                      <button
                        className="isotope-filters-toggle isotope-filters-toggle-1 button"
                        data-custom-toggle="#isotope-filters"
                        data-custom-toggle-disable-on-blur="true"
                      >
                        Filter
                        <span
                          className="caret"
                        />
                      </button>
                      <WorldSectionLinks
                        changeSection={[Function]}
                      />
                    </div>,
                  ],
                  "className": "section-title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "World",
                    },
                    "ref": null,
                    "rendered": "World",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <button
                          className="isotope-filters-toggle isotope-filters-toggle-1 button"
                          data-custom-toggle="#isotope-filters"
                          data-custom-toggle-disable-on-blur="true"
                        >
                          Filter
                          <span
                            className="caret"
                          />
                        </button>,
                        <WorldSectionLinks
                          changeSection={[Function]}
                        />,
                      ],
                      "className": "isotope-filters isotope-filters-horizontal",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Filter",
                            <span
                              className="caret"
                            />,
                          ],
                          "className": "isotope-filters-toggle isotope-filters-toggle-1 button",
                          "data-custom-toggle": "#isotope-filters",
                          "data-custom-toggle-disable-on-blur": "true",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Filter",
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "caret",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "span",
                          },
                        ],
                        "type": "button",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "changeSection": [Function],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeNews": Array [
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 8,
                    "section": "World",
                    "title": "news title1",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 9,
                    "section": "World",
                    "title": "news title2",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 10,
                    "section": "World",
                    "title": "news title3",
                  },
                  Object {
                    "author": "Jon Snow",
                    "date": "28-01-2019",
                    "description": "this is the description",
                    "id": 11,
                    "section": "World",
                    "title": "news Business",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
